### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by harshitacodes**
## Short description of the changes
- Add missing hyphens, like `open source -> open-source`
- Fix spellings and punctuations

:sparkles: :cake:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Jaeger - a Distributed Tracing System
 
 Jaeger, inspired by [Dapper][dapper] and [OpenZipkin](https://zipkin.io),
-is a distributed tracing system released as open source by [Uber Technologies][ubeross].
+is a distributed tracing system released as an open-source by [Uber Technologies][ubeross].
```
Comment (yurishkuro) [inline README.md]:
```suggestion
is a distributed tracing platform released as an open-source by [Uber Technologies][ubeross].
```

File: README.md [hunk 2]
```diff
@@ -33,28 +33,28 @@ For example, any given Jaeger installation at Uber is typically processing sever
...
 
 ### Multiple storage backends
 
-Jaeger supports two popular open source NoSQL databases as trace storage backends: Cassandra 3.4+ and Elasticsearch 5.x/6.x.
+Jaeger supports two popular open-source NoSQL databases as trace storage backends: Cassandra 3.4+ and Elasticsearch 5.x/6.x.
```
Comment (yurishkuro) [inline README.md]:
```suggestion
Jaeger supports two popular open-source NoSQL databases as trace storage backends: Cassandra and Elasticsearch.
```

File: README.md [hunk 3]
```diff
@@ -70,11 +70,11 @@ also supported). Logs are written to standard out using the structured logging l
 
 Third-party security audits of Jaeger are available in https://github.com/jaegertracing/security-audits. Please see [Issue #1718](https://github.com/jaegertracing/jaeger/issues/1718) for the summary of available security mechanisms in Jaeger.
 
-### Backwards compatibility with Zipkin
+### Backward compatibility with Zipkin
```
Comment (yurishkuro) [inline README.md]: both are correct

**Comment by JStickler** [comment]
I know that the rules of grammar say to use a hyphen between two words when you're using them as an adjective, but in this case I don't think it's necessarily wrong to say "Open Source"  since the OSI (https://opensource.org/) doesn't use a hyphen and open source without the hyphen is a pretty common usage.

**Comment by yurishkuro** [comment]
@harshitacodes please make sure all commits are signed and pass the DCO check (you may have to squash them if you missed sign-off on earlier commits).

**Pull request merged**